modules/platform/nuxeo-easyshare-core/src/main/java/org/nuxeo/easyshare/EasyShareEncryptionUtil.java

```diff
@@ -0,0 +1,180 @@
+package org.nuxeo.easyshare;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.PBEParameterSpec;
+
+import org.apache.commons.codec.binary.Base64;
+
+public class EasyShareEncryptionUtil {
+
+	// some random salt
+	private static final byte[] SALT = { (byte) 0x21, (byte) 0x21, (byte) 0xF0, (byte) 0x55, (byte) 0xC3, (byte) 0x9F,
+			(byte) 0x5A, (byte) 0x75 };
+
+	private final static int ITERATION_COUNT = 31;
+
+	private EasyShareEncryptionUtil() {
+	}
+
+	/**
+	 * Return a predefined SALT
+	 * 
+	 * @return Predefined SAL (@see SALT)
+	 */
+	public static String getDefaultSaltAsString() {
+
+		return Base64.encodeBase64String(SALT);
+	}
+
+	/**
+	 * Generate a custom salt as base64 encoded string
+	 * 
+	 * @return random base64 string
+	 */
+	public static String getNextSaltAsString() {
+
+		return Base64.encodeBase64String(getNextSaltAsByte());
+	}
+
+	/**
+	 * Encode an input string with passed salt string (default value will be
+	 * used for salt if param is empty/null) (wrapper for byte[] salt)
+	 * 
+	 * @param input
+	 *            String to encode
+	 * @param salt
+	 *            SALT to use
+	 * @return An encoded string
+	 */
+	public static String encodeWithSaltString(String input, String salt) {
+		try {
+			return encodeWithSaltByte(input, Base64.decodeBase64(salt));
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
+	}
+
+	/**
+	 * Decode an encoded token with passed salt string (default value will be
+	 * used for salt if param is empty/null)
+	 * 
+	 * @param token
+	 *            Encoded String to decode using passed salt
+	 * @param salt
+	 *            SALT to use when decoding
+	 * @return Decoded string from token
+	 */
+	public static String decodeWithSaltString(String token, String salt) {
+		try {
+			return decodeWithSaltByte(token, Base64.decodeBase64(salt));
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
```

Comment:
When does it happen? Shouldn't we re-throw a NuxeoException with a Not found/Denied code?

Reply:
Changed it to remove the try/catch (managed in decodeWithSaltByte throwing NuxeoException